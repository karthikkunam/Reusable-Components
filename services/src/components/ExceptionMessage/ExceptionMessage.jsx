import React from 'react';
import PropTypes from 'prop-types';

function ExceptionMessage(props) {
    const { messageType, messageHeader, messageInfo } = props;

    let messageLogo;
    if (messageType === 'success') {
        messageLogo = <span className="icon icon-status-checkmark-reversed active" />;
    } else if (messageType === 'error') {
        messageLogo = <span className="icon icon-status-error-reversed active" />;
    } else {
        messageLogo = <span className="icon icon-status-info-reversed" />;
    }

    const msgs = messageInfo.map(msg => (<div key={msg}> {msg} </div>));
    return (
        <div className="alert exception-message">
            <div> {messageLogo} </div>
            <div className="header"> {messageHeader} </div>
            <div className="body"> {msgs} </div>
        </div>
    );
}

ExceptionMessage.displayName = 'ExceptionMessage';

ExceptionMessage.propTypes = {
    messageType: PropTypes.oneOf(['success', 'info', 'error']),
    messageHeader: PropTypes.string,
    messageInfo: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string])),
};

ExceptionMessage.defaultProps = {
    messageType: 'info',
    messageInfo: ['No data to display.']
};

export default ExceptionMessage;