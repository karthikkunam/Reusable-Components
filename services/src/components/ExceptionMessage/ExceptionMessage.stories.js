import React from 'react';
import ExceptionMessage from '.';

export default {
    title: 'ExceptionMessage',
    component: ExceptionMessage,
};

const successMessage = ['All deliveries have been checked-in. Good job!'];
const infoMessage = ['There are no deliveries available for check-in at this time'];
const errorMessage = ['There seems to be a problem with the network.', 'Please check your connection and try again.'];

export const SuccessMessage = () => (
    <ExceptionMessage
        messageType="success"
        messageHeader="All Available Deliveries Checked-in"
        messageInfo={successMessage}
    />
);

export const InfoMessage = () => (
    <ExceptionMessage
        messageType="info"
        messageHeader="No Deliveries"
        messageInfo={infoMessage}
    />
);
export const ErrorMessage = () => (
    <ExceptionMessage
        messageType="error"
        messageHeader="Connection Error"
        messageInfo={errorMessage}
    />
);
