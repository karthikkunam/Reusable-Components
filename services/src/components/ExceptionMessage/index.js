import ExceptionMessage from './ExceptionMessage';

export default ExceptionMessage;