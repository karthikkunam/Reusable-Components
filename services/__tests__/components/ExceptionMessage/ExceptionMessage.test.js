import React from 'react';
import { shallow } from 'enzyme';
import ExceptionMessage from '../../../src/components/ExceptionMessage';

describe('ExceptionMessage Component', () => {
  it('renders correctly', () => {
    const component = shallow(<ExceptionMessage/>);
    expect(component).toMatchSnapshot();
  });
  it('renders default props correctly', () => {
      const props ={
        messageType: 'info', 
        messageHeader: '', 
        messageInfo: ['']
      };
    const component = shallow(<ExceptionMessage {...props}/>);
    expect(component).toMatchSnapshot();
  });
  it('renders success message correctly', () => {
      const props ={
        messageType: 'success', 
        messageHeader: '', 
        messageInfo: ['']
      };
    const component = shallow(<ExceptionMessage {...props}/>);
    expect(component).toMatchSnapshot();
  });
  it('renders error message correctly', () => {
      const props ={
        messageType: 'error', 
        messageHeader: '', 
        messageInfo: ['']
      };
    const component = shallow(<ExceptionMessage {...props}/>);
    expect(component).toMatchSnapshot();
  });
});