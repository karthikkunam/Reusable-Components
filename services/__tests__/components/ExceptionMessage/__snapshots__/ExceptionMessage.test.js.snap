// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExceptionMessage Component renders correctly 1`] = `
<div
  className="alert exception-message"
>
  <div>
     
    <span
      className="icon icon-status-info-reversed"
    />
     
  </div>
  <div
    className="header"
  >
     
     
  </div>
  <div
    className="body"
  >
     
    <div
      key="No data to display."
    >
       
      No data to display.
       
    </div>
     
  </div>
</div>
`;

exports[`ExceptionMessage Component renders default props correctly 1`] = `
<div
  className="alert exception-message"
>
  <div>
     
    <span
      className="icon icon-status-info-reversed"
    />
     
  </div>
  <div
    className="header"
  >
     
     
  </div>
  <div
    className="body"
  >
     
    <div
      key=""
    >
       
       
    </div>
     
  </div>
</div>
`;

exports[`ExceptionMessage Component renders error message correctly 1`] = `
<div
  className="alert exception-message"
>
  <div>
     
    <span
      className="icon icon-status-error-reversed active"
    />
     
  </div>
  <div
    className="header"
  >
     
     
  </div>
  <div
    className="body"
  >
     
    <div
      key=""
    >
       
       
    </div>
     
  </div>
</div>
`;

exports[`ExceptionMessage Component renders success message correctly 1`] = `
<div
  className="alert exception-message"
>
  <div>
     
    <span
      className="icon icon-status-checkmark-reversed active"
    />
     
  </div>
  <div
    className="header"
  >
     
     
  </div>
  <div
    className="body"
  >
     
    <div
      key=""
    >
       
       
    </div>
     
  </div>
</div>
`;
